lib/broadside/deploy/ecs.rb

```diff
@@ -124,39 +128,58 @@ def bash
 
     private
 
-    def get_running_instance_ip_at_index(index)
-      ips = get_running_instance_ips
-      ips.size > index ? ips[index] : false
-    end
-
-    def get_running_instance_ip_at_index!(index)
-      ip = get_running_instance_ip_at_index(index)
-      return ip if ip
-      exception 'Instance index greater than available instance count.'
-    end
-
     # removes latest n task definitions
     def deregister_tasks(count)
-      get_task_def_ids.last(count).each do |td_id|
+      get_task_definition_arns.last(count).each do |td_id|
         ecs_client.deregister_task_definition({task_definition: td_id})
         debug "Deregistered #{td_id}"
       end
     end
 
     # creates a new task revision using current directory's env vars and provided tag
     def update_task
-      debug "Creating a new task definition..."
-      new_task_def = create_new_task_revision
+      unless get_latest_task_def_id
+        exception "No first task definition and cannot create one" unless @task_definition_config
 
-      container_def = new_task_def[:container_definitions].select { |c| c[:name] == family }.first
-      container_def[:environment] = @deploy_config.env_vars
-      container_def[:image] = image_tag
-      container_def[:command] = @deploy_config.command
+        info "Creating an initial task definition from the config..."
+        create_task_definition(family)
+      end
 
-      new_task_def_id = ecs_client.register_task_definition(new_task_def).task_definition.task_definition_arn
+      debug "Creating a new task definition..."
+      new_task_def_id = ecs_client.register_task_definition(create_new_task_revision).task_definition.task_definition_arn
       debug "Successfully created #{new_task_def_id}"
     end
 
+    def create_service(name, options = {})
+      ecs_client.create_service(
+        {
+          cluster: config.ecs.cluster,
+          desired_count: 0,
+          service_name: name,
+          task_definition: name
+        }.deep_merge(options)
+      )
+    end
+
+    def create_task_definition(name, options = {})
+      raise ArgumentError, 'No :image provided!' unless options[:container_definitions].try(:first).try(:[], [:image])
+
+      ecs.register_task_definition(
+        {
+          container_definitions: [
+            {
+              name: name,
+              command: @command,
+              cpu: 1,
+              essential: true,
```

Comment:
i thought about that.  you can't have a "constant" really because a  name, command, family are all dynamic.  so i thought about making building the config a separate method but really having a separate method is _almost_ the same as having a method with one line call that builds the aws definition hash.